arm-asazure/2016-05-06/swagger/asazure.json

```diff
@@ -0,0 +1,551 @@
+      "get": {
+        "description": "Gets result of the specified operation. The API return the HTTP status code and a response body",
+        "tags": [ "RpProxy" ],
+        "operationId": "RpProxy_GetOperationStatus",
+        "parameters": [
+          {
+            "name": "operationId",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "x-ms-skip-url-encoding": true,
+            "description": "Identifier for the operation for which to get status"
+          },
+          {
+            "name": "location",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "x-ms-skip-url-encoding": true,
+            "description": "Region in which the server operation was executed"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": { "$ref": "#/definitions/OperationDetail" }
+          },
+          "202": {
+            "description": "Accepted",
+            "schema": { "$ref": "#/definitions/OperationDetail" }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.AnalysisServices/locations/{location}/operationresults/{operationId}": {
+      "get": {
+        "description": "Gets result of the specified operation. The API return only the HTTP status code and does not have a response body",
+        "tags": [ "RpProxy" ],
+        "operationId": "RpProxy_GetOperationResult",
+        "parameters": [
+          {
+            "name": "operationId",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "x-ms-skip-url-encoding": true,
+            "description": "Identifier for the operation for which to get status"
+          },
+          {
+            "name": "location",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "x-ms-skip-url-encoding": true,
+            "description": "Region in which the server operation was executed"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          },
+          "202": {
+            "description": "Accepted"
+          },
+          "204": {
+            "description": "No Content"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "RpServerUpdateRequest": {
+      "description": "Update request specification",
+      "type": "object",
+      "properties": {
+        "location": {
```

Comment:
Which indicates that,
> Although location cannot be updated (must have the same value as provided in create), it is a required property. This is an anti-pattern for the body parameter of PATCH operation

If the above statement is not followed then it is highly likely that the customer will not provide "location" in update operation and the operation will fail. He will know about this by trial and error.

Our goal is to "Direct the customer to the correct path, so that he gets it right in the first attempt; without leaving him frustrated."

@athipp - It would be nice if you can:
- add this to the description of the location property.
    "Location of the service. Cannot be changed to a different value than the value used during creation time"
- add location as a "required" property (don't like this, but cant help).

/cc @johanste, @anudeep, #selvasingh - What are your thoughts on this?